tests/framework/PostProcessors/BasicStatistics/gold/basicStatsMonteCarloAnalytic/analyticalTest_basicStatPP_dump.csv

```diff
@@ -1,2 +1,2 @@
 skew_x,skew_y,vc_x,vc_y,percentile_5_x,percentile_95_x,percentile_5_y,percentile_95_y,mean_x,mean_y,kurt_x,kurt_y,median_x,median_y,max_x,max_y,min_x,min_y,samp_x,samp_y,var_x,var_y,sigma_x,sigma_y,nsen_x_x,nsen_x_y,nsen_y_x,nsen_y_y,sen_x_x,sen_x_y,sen_y_x,sen_y_y,pear_x_x,pear_x_y,pear_y_x,pear_y_y,cov_x_x,cov_x_y,cov_y_x,cov_y_y,vsen_x_x,vsen_x_y,vsen_y_x,vsen_y_y
--0.0691572462104,-0.0180192261137,0.491941113434,0.46467884342,17.732771122,181.912710136,25.5790457803,184.240434035,102.005526367,105.895162869,-0.139930546516,0.184012286626,100.962661272,105.082835069,256.130010971,291.892206598,-41.8819801615,-67.3356620259,1000.0,1000.0,2518.10387863,2421.35264613,50.1807122173,49.2072418057,1.0,-0.0477758254403,-0.042627293499,1.0,1.0,-0.0460209710208,-0.0442527414791,1.0,1.0,-0.0451281966535,-0.0451281966535,1.0,2518.10387863,-111.432999959,-111.432999959,2421.35264613,1.0,-0.0460209710208,-0.0442527414791,1.0
```

Comment:
@dgarrett622 I have a question here. Since you have switched the calculation to "midpoint", why the calculation is different from the current devel version? I mean why do we need to regold it also?

Reply:
@wangcj05 I looked a little more into this discrepancy. When using _computeWeightedPercentile for a variable with N values, the smallest value is appended with weight zero. Adding in this additional value appears to cause the discrepancy with xarray. _computeWeightedPercentile works with N+1 values where xarray works with N values. For many requested percentiles both methods return the same value. However because the two methods use arrays of different length, they may return different values. I get the same value for the 5th percentile using each method and differing values for the 95th percentile when using the data from this test. As another example, when requesting a very small percentile (1e-9) _computeWeightedPercentile will return the smallest value where xarray returns the average of the two smallest values.

The differences in this particular example are small, on the order of 0.06% difference between the values. As written, the majority of cases will be calculated using xarray. The only time _computeWeightedPercentile would be used is if the probability weights are not all the same (xarray does not currently support weighted quantile calculation, I think it is because numpy does not have it implemented). I still think using the standard library xarray wherever possible is the best way forward.

Reply:
Sounds good to me. Thanks for checking it.